.travis.yml

```diff
@@ -7,37 +7,48 @@ notifications:
       - "irc.freenode.net#haskell-india"
     use_notice: true
 
-# Setting up caches
-before-cache:
-  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/build-reports.log
-  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/00-index.tar
-
 cache:
   directories:
-    - $HOME/.cabsnap
-    - $HOME/.cabal/packages
+    - $HOME/.ghc   
+    - $HOME/.cabal
+    - $HOME/.stack
 
+before_install:
+# Download and unpack the stack executable
+- mkdir -p ~/.local/bin
+- export PATH=$HOME/.local/bin:$PATH
+- travis_retry curl -L https://www.stackage.org/stack/linux-x86_64 | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
```

Comment:
If fpcomplete repos are available then we can also use addons toget the stack executable.